docker-stack.yml

```diff
@@ -0,0 +1,74 @@
+version: '3.3'
+
+networks:
+  net:
+    driver: overlay
+
+configs:
+  elastic_config:
+    file: ./elasticsearch/config/elasticsearch.yml
+  logstash_config:
+    file: ./logstash/config/logstash.yml
+  logstash_pipeline:
+    file: ./logstash/pipeline/logstash.conf
+  kibana_config:
+    file: ./kibana/config/kibana.yml
+
+services:
+  elasticsearch:
+    image: docker.elastic.co/elasticsearch/elasticsearch-oss:6.2.1
+    ports:
+      - "9200:9200"
+      - "9300:9300"
+    configs:
+      - source: elastic_config
+        target: /usr/share/elasticsearch/config/elasticsearch.yml
+        uid: '1000'
+        gid: '0'
+    environment:
+      ES_JAVA_OPTS: "-Xmx256m -Xms256m"
+    networks:
+      - net
+    deploy:
+      mode: replicated
+      replicas: 1
+
+  logstash:
+    image: docker.elastic.co/logstash/logstash-oss:6.2.1
+    ports:
+      - "5000:5000"
+    configs:
+      - source: logstash_config
+        target: /usr/share/logstash/config/logstash.yml
+      - source: logstash_pipeline
+        target: /usr/share/logstash/pipeline/logstash.conf
+    environment:
+      LS_JAVA_OPTS: "-Xmx256m -Xms256m"
+    networks:
+      - net
+    deploy:
+      mode: replicated
+      replicas: 1
+      resources:
```

Comment:
I don't know if the resources limit are actually required here. This project aims to provide a generic solution, we might want to let the user actually define these.

Reply:
You have JAVA limitations passed in via environment variables.  Wouldn't it make sense to provide something similar for the docker container itself?   Users can easily edit or remove these but if you're new to docker as I am you may not even know they exist without seeing them there.